### ISSUE TITLE
CA1861 exclude static ReadOnlyCollection<T> field/properties

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn-analyzers#6629]
### Analyzer `CA1861 Avoid constant arrays as arguments`

**Diagnostic ID**: [CA1861](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1716)

### Describe the improvement

We have excluded static readonly fields/properties instantiation, we also need to exclude `static ReadOnlyCollection<T>` field/property  initializtions

Example: 
```cs
public class Test
{
    private static ReadOnlyCollection<string>? s_errorPayloadNames;

    private void M(string eventName, string msg)
    {
        var eventCallbackArgs = new EventWrittenEventArgs(this, 0)
        {
            EventName = eventName,
            Message = msg,
            Payload = new ReadOnlyCollection<object?>(new object[] { msg }),
            // Should not warn here
            PayloadNames = s_errorPayloadNames ??= new ReadOnlyCollection<string>(new string[] { "message" })
        };
    }
}
```

### Additional context

Related to https://github.com/dotnet/runtime/pull/86229#discussion_r1193870738

**Comment by stephentoub** [comment]
Why would we handle ReadOnlyCollection specially?  Why not, for example, instead not warn if the array is part of an expression used to set a static field?

**Comment by buyaa-n** [comment]
> Why would we handle ReadOnlyCollection specially? 

We have excluded static readonly fields/properties instantiation because that instantiation will be done only once. Same reason apply for static `ReadOnlyCollection`. 

> Why not, for example, instead not warn if the array is part of an expression used to set a static field?

Isn't that expression can run multiple times?
